src/hdmf/container.py

```diff
@@ -85,10 +86,82 @@ def setter(self, val):
-            self.fields[name] = val
+            self.fields[name] = self._field_config(arg_name=name, val=val)
 
         return setter
 
+    def get_data_type(self):
+        """
+        Method that return data type.
+        """
+        return getattr(self, self._data_type_attr)
+
+
+    def _field_config(self, arg_name, val):
+        """
+        This method will be called in the setter. The termset configuration will be used (if loaded)
+        to check for a defined TermSet associated with the field. If found, the value of the field
+        will be wrapped with a TermSetWrapper.
+
+        Even though the path field in the configurator can be a list of paths, the config
+        itself is only one file. When a user loads custom configs, the config is appended/modified.
+        The modifications are not written to file, avoiding permanent modifications.
+        """
+        # load termset configuration file from global Config
+        from hdmf.common import get_type_map # circular import
+        type_map = get_type_map()
+        configurator = type_map.ts_config
+
+        if len(configurator.path)>0:
+            # The type_map has a config always set; however, when toggled off, the config path is empty.
+            CUR_DIR = os.path.dirname(os.path.realpath(configurator.path[0]))
+            termset_config = configurator.config
+        else:
+            return val
+
+        # check to see that the namespace for the container is in the config
+        if self.namespace not in type_map.container_types:
+            msg = "%s not found within loaded configuration." % self.namespace
+            warn(msg)
+            return val
+        else:
+            # check to see that the container type is in the config under the namespace
+            config_namespace = termset_config['namespaces'][self.namespace]
+            data_type = self.get_data_type()
+
+            if data_type not in config_namespace['data_types']:
+                msg = '%s not found within the configuration for %s' % (data_type, self.namespace)
+                warn(msg)
+                return val
+            else:
+                for attr in config_namespace['data_types'][data_type]:
+                    obj_mapper = type_map.get_map(self)
+
+                    # get the spec according to attr name in schema
+                    # Note: this is the name for the field in the config
+                    spec = obj_mapper.get_attr_spec(attr)
+
+                    # In the case of dealing with datasets directly or not defined in the spec.
+                    # (Data/VectorData/DynamicTable/etc)
+                    if spec is None:
+                        msg = "Spec not found for %s." % attr
+                        warn(msg)
+                        return val
+                    else:
+                        # If the val has been manually wrapped then skip checking the config for the attr
+                        if type(val) == TermSetWrapper:
```

Comment:
```suggestion
                        if isinstance(val, TermSetWrapper):
```
This is more general and allows for subclassing of `TermSetWrapper`.